types/views/views.go

```diff
@@ -170,6 +170,23 @@ func SliceContains[T comparable](v Slice[T], e T) bool {
 	return false
 }
 
+// SliceEquals reports whether a and b contain the same elements, regardless of order.
+func SliceEquals[T comparable](a, b Slice[T]) bool {
```

Comment:
this sounds too much like https://pkg.go.dev/golang.org/x/exp/slices#Equal which is NOT this function.

Let's name this something uglier like `SliceEqualAnyOrder`

Then we can reserve the name `SliceEqual` for one that just forwards to x/exp/slices.Equal.

Clarify the nilness of the slice doesn't matter too? (unlike reflect.DeepEqual)